drivers/console/uart_mcumgr.c

```diff
@@ -111,6 +118,45 @@ static struct uart_mcumgr_rx_buf *uart_mcumgr_rx_byte(uint8_t byte)
 	return NULL;
 }
 
+#if defined(CONFIG_MCUMGR_SMP_UART_ASYNC)
+static void uart_mcumgr_async(const struct device *dev, struct uart_event *evt, void *user_data)
+{
+	struct uart_mcumgr_rx_buf *rx_buf;
+	uint8_t *p;
+	int len;
+	ARG_UNUSED(dev);
+
+	switch (evt->type) {
+	case UART_TX_DONE:
+		break;
+	case UART_TX_ABORTED:
+		break;
+	case UART_RX_RDY:
+		len = evt->data.rx.len;
+		p = &evt->data.rx.buf[evt->data.rx.offset];
+
+		for (int i = 0; i < len; i++) {
+			rx_buf = uart_mcumgr_rx_byte(p[i]);
+			if (rx_buf != NULL) {
+				uart_mgumgr_recv_cb(rx_buf);
+			}
+		}
+		break;
+	case UART_RX_BUF_RELEASED:
+		break;
+	case UART_RX_DISABLED:
```

Comment:
you can reset `async_current` to 0 here. In case mcumgr is sharing uart and may be re-enabled.

Reply:
OK